api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -173,4 +177,92 @@ public void deleteOrderGroupAttributeType_shouldDeleteOrderGroupAttributeTypeFro
 		dao.deleteOrderGroupAttributeType(orderGroupAttributeType);
 		assertNull(dao.getOrderGroupAttributeByUuid(UUID));
 	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getAllOrderAttributeTypes()}
+	 * @throws Exception
+	 */
+	@Test
+	public void getAllOrderAttributeTypes_shouldGetAllOrderAttributeTypes() {
+		List<OrderAttributeType> orderAttributeTypes = dao.getAllOrderAttributeTypes();
+		assertEquals(orderAttributeTypes.size(), 4);
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderAttributeTypeByUuid(String)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderAttributeTypeByUuid_shouldGetOrderAttributeTypeByUuidGivenTheUuid() {
+		OrderAttributeType orderAttributeType = dao.getOrderAttributeTypeByUuid("9758d106-79b0-4f45-8d8c-ae8b3f25d72a");
+		assertEquals("Referral", orderAttributeType.getName());
+	}
+	
+	@Test
+	public void saveOrderAttributeType_shouldSaveOrderAttributeTypeToDatabase() {
+		int initialOrderAttributeTypeCount = dao.getAllOrderAttributeTypes().size();
+		final Order order = Context.getOrderService().getOrder(1);
+		OrderAttributeType orderAttributeType = new OrderAttributeType();
+		orderAttributeType.setOrderAttributeTypeId(5);
+		orderAttributeType.setName("External Referral");
+		orderAttributeType.setMinOccurs(5);
+		orderAttributeType.setCreator(order.getCreator());
+		orderAttributeType.setDateCreated(order.getDateCreated());
+		orderAttributeType.setRetired(false);
+		orderAttributeType.setUuid("81b95c51-865b-48c6-aacf-cc8f21e69f5e");
+
+		for (OrderAttribute orderAttribute : order.getActiveAttributes()) {
+			if (order.getOrderId() == 1) {
+				orderAttribute.setAttributeType(orderAttributeType);
+			}
+		}
+		OrderAttributeType orderAttributeTypeToSave = dao.saveOrderAttributeType(orderAttributeType);
+		assertNotNull(orderAttributeTypeToSave.getOrderAttributeTypeId(), "Saved OrderAttribute Type");
+		assertEquals(initialOrderAttributeTypeCount + 1, dao.getAllOrderAttributeTypes().size());
+	}
+
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderAttributeTypeById(Integer)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderAttributeTypeById_shouldFailGivenNullAsId() {
```

Comment:
I'd name it as `getOrderAttributeTypeById_shouldThrowApiExceptionIfGivenIdIsNull()`